lesson05/main.js

```diff
@@ -1,17 +1,24 @@
-const ul = document.getElementById("js-ul"); 
-const fragment = document.createDocumentFragment(); 
 const elements = [ 
   {to: "bookmark.html", img:"img/1.png", alt:"画像1", text: "ブックマーク"}, 
   {to: "message.html", img:"img/2.png", alt:"画像2", text: "メッセージ"} 
 ]; 
-elements.forEach(function(element){ 
-  const li = document.createElement("li"); 
-  const a = document.createElement("a"); 
-  const img = document.createElement("img"); 
-  a.href = element.to; 
-  a.textContent = element.text; 
-  img.src = element.img; 
-  img.alt = element.alt; 
-  fragment.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin",img); 
-}); 
-ul.appendChild(fragment);
+
+let promise =  new Promise((ok)=> {
+  ok(elements);
+});
+
+promise.then((elements)=> {

```

Comment:
[ IMO ]
You can write like this
```suggestion
const promise =  new Promise((resolve)=> {
  resolve(elements);
}).then((elements)=> {

```
- I see no reason to use let.
- I think that the argument to promise should be written in `resolve`, I have never seen [MDN](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Promise) or anything else written with an argument other than resolve.
(If you have any references, I'd be glad to hear them.)
- Promise can chain` then`.

Reply:
Thank you haruna-san :)
you gave me a lot of advices.

It's simple.
I'll change my code !